Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStoreConfiguration.cs

```diff
@@ -203,32 +205,62 @@ public DistributedCentralStoreConfiguration(AbsolutePath cacheRoot)
         public int MaxSimultaneousCopies { get; set; } = 10;
     }
 
+    /// <summary>
+    /// Provides Azure Storage authentication options for <see cref="BlobCentralStorage"/>
+    /// </summary>
+    public class AzureStorageCredentials
+    {
+        /// <nodoc />
+        public string ConnectionString { get; private set; } = null;
+
+        /// <nodoc />
+        public StorageCredentials StorageCredentials { get; private set; } = null;
+
+        public AzureStorageCredentials(string connectionString)
+        {
+            Contract.Requires(connectionString != null);
+            ConnectionString = connectionString;
+        }
+
+        public AzureStorageCredentials(StorageCredentials storageCredentials)
+        {
+            Contract.Requires(storageCredentials != null);

```

Comment:
Add this here: Contract.Requires(credentials.StorageCredentials.IsSAS);

Reply:
Are we sure that's what we want? not requiring this would allow us to also change passwords while still running, for example.